frontend/src/components/elements/DeckGlChart/DeckGlChart.tsx

```diff
@@ -96,33 +97,73 @@ class DeckGlChart extends React.PureComponent<PropsWithHeight, State> {
     setTimeout(this.fixHexLayerBug_bound, 0)
   }
 
+  updateInitialViewState = (): void => {
+    const specStr = this.props.element.get("spec")
+    const spec = specStr ? JSON.parse(specStr) : {}
+    const v = spec.viewport || {}
+    const { width, height } = this.props
+    this.initialViewState = {
+      width: v.width || width,
+      height: v.height || height,
+      longitude: v.longitude || 0,
+      latitude: v.latitude || 0,
+      pitch: v.pitch || 0,
+      bearing: v.bearing || 0,
+      zoom: v.zoom || 1,
+    }
+
+    this.mapStyle = getStyleUrl(v.mapStyle)
+  }
+
   fixHexLayerBug(): void {
     this.setState({ initialized: true })
   }
 
   render(): JSX.Element {
+    this.updateInitialViewState()
     return (
       <div
-        className="deckglchart stDeckGlChart"
+        className="deckglchart"
         style={{
-          height: this.initialViewState.height,
+          height: this.initialViewState.height + 60,
           width: this.initialViewState.width,
         }}
       >
-        <DeckGL
-          initialViewState={this.initialViewState}
-          height={this.initialViewState.height}
-          width={this.initialViewState.width}
-          controller
-          layers={this.state.initialized ? this.buildLayers() : []}
+        <div
+          className="stDeckGlChart"
+          style={{
+            height: this.initialViewState.height,
+            width: this.initialViewState.width,
+          }}
         >
-          <StaticMap
+          <DeckGL
+            initialViewState={this.initialViewState}
             height={this.initialViewState.height}
             width={this.initialViewState.width}
-            mapStyle={this.mapStyle}
-            mapboxApiAccessToken={MAPBOX_ACCESS_TOKEN}
-          />
-        </DeckGL>
+            controller
+            layers={this.state.initialized ? this.buildLayers() : []}
+          >
+            <StaticMap
+              height={this.initialViewState.height}
+              width={this.initialViewState.width}
+              mapStyle={this.mapStyle}
+              mapboxApiAccessToken={MAPBOX_ACCESS_TOKEN}
+            />
+          </DeckGL>
+        </div>
+        <div
+          className={"alert alert-warning stText"}
+          style={{ width: this.initialViewState.width }}
+        >
+          <div className="markdown-text-container">
+            <StreamlitMarkdown
```

Comment:
Instead of adding this warning via JS, it's best to do in Python inside `DeltaGenerator.deck_gl_chart`:

```py
def deck_gl_chart(...):
  st.warning("""
    WARNING GOES HERE
    """)
  # rest of the code goes here
```

Then we can also easily add a keyword parameter that turns the warning off (which is something we should totally do!):

```py
def deck_gl_chart(..., suppress_deprecation_warning=False):
  if not suppress_deprecation_warning:
    st.warning("""
      WARNING GOES HERE
      """)
  # rest of the code goes here
```

(then you can also undo all changes to this file)